docs/sdk-writer-guide.md

```diff
@@ -13,25 +13,33 @@ These SDK writer output contains just a carbondata and carbonindex files. No met
  
  import org.apache.carbondata.common.exceptions.sql.InvalidLoadOptionException;
  import org.apache.carbondata.core.metadata.datatype.DataTypes;
+ import org.apache.carbondata.core.util.CarbonProperties;
  import org.apache.carbondata.sdk.file.CarbonWriter;
  import org.apache.carbondata.sdk.file.CarbonWriterBuilder;
  import org.apache.carbondata.sdk.file.Field;
  import org.apache.carbondata.sdk.file.Schema;
  
  public class TestSdk {
- 
+
+   // pass true or false while executing the main to use offheap memory or not
    public static void main(String[] args) throws IOException, InvalidLoadOptionException {
-     testSdkWriter();
+     if(args[0] != null) {
```

Comment:
It will throw "Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 0
	at org.apache.carbondata.examples.sdk.TestSDK.main(TestSDK.java:17)"

Reply:
BufferedReader reader = 
                   new BufferedReader(new InputStreamReader(System.in));
        String enableOffheapSortVal = reader.readLine();

Reply:
is this ok?

Reply:
Have you run this test case in you machine? I run it but failed

Reply:
Didn't you rebase to the latest master?

Reply:
it may be failing because of invalid path given in testSdkWriter().

![image](https://user-images.githubusercontent.com/14244942/40309994-0ebc1d0c-5d29-11e8-8b1f-90898bdbe783.png)

Reply:
yeah, this path also has problem. I raised a commit just now:https://github.com/xubo245/carbondata/commit/09f387717e5eabb01bf9f688c733e3342ce22799, it can run success, maybe you can refer